go/multipooler/poolerserver/pooler.go

```diff
@@ -17,119 +17,180 @@ package poolerserver
 
 import (
 	"context"
-	"database/sql"
 	"fmt"
 	"log/slog"
+	"sync"
+	"time"
 
 	"github.com/multigres/multigres/go/multipooler/executor"
-	"github.com/multigres/multigres/go/multipooler/manager"
 	"github.com/multigres/multigres/go/multipooler/queryservice"
-	"github.com/multigres/multigres/go/servenv"
+	clustermetadatapb "github.com/multigres/multigres/go/pb/clustermetadata"
 )
 
 // MultiPooler is the core pooler implementation
+// It implements the PoolerController interface
```

Comment:
We should probably expland on this comment, to write out more explicitly what the purpose of this is. And also we should probably rename the struct. MultiPooler doesn't seem like a very descriptive name